lib/Cron/Task.php

```diff
@@ -32,12 +109,116 @@ public function __construct() {
 	 * @return void
 	 */
 	protected function run($argument) {
-		if (!\OCP\App::isEnabled('files_antivirus')) {
+		if (!\OCP\App::isEnabled('files_antivirus')
+			|| $this->appConfig->getAvScanBackground() !== 'true'
+		) {
+			return;
+		}
+
+		// locate files that are not checked yet
+		try {
+			$result = $this->getFilesForScan();
+		} catch (\Exception $e) {
```

Comment:
I guess the exceptions here will be mostly from the DB layer... I'm not sure if this is one of those cases where catching every exception is fine

Reply:
it's an 3rd party application cron job. Any exception inside it is probably related to the app and should be logged by the app itself.